src/main/kotlin/org/opensearch/replication/ReplicationPlugin.kt

```diff
@@ -359,7 +363,14 @@ internal class ReplicationPlugin : Plugin(), ActionPlugin, PersistentTaskPlugin,
 
     override fun getEngineFactory(indexSettings: IndexSettings): Optional<EngineFactory> {
         return if (indexSettings.settings.get(REPLICATED_INDEX_SETTING.key) != null) {
-            Optional.of(EngineFactory { config -> ReplicationEngine(config) })
+            Optional.of(EngineFactory { config ->
+                // Use NRTSegmentReplicationEngine for SEGMENT replication type indices replica shards
+                if (config.isReadOnlyReplica && indexSettings.settings.get(INDEX_REPLICATION_TYPE_SETTING.key) != null && indexSettings.settings.get(INDEX_REPLICATION_TYPE_SETTING.key).equals("SEGMENT")) {
```

Comment:
- we can drop explicit `null` check and use "?" instead for obtaining replication type setting. 
- Can we use constant for "SEGMENT" (from the allowed values)?

Reply:
Looks like `config.isReadOnlyReplica` already have segrep settings check inside EngineConfig. Removing segrep setting explicit check. 

```
    public boolean isReadOnlyReplica() {
        return indexSettings.isSegRepEnabled() && isReadOnlyReplica;
    }
```